### PR TITLE
chore/remove globals

### PR DESCRIPTION
**Pull request by alex-courtis**
First round. No significant refactoring. I'll do some more when I get the energy.

Second round, merge after this: #1280

Final round, merge after #1280 : #1281

### CODE REVIEW

File: lua/nvim-tree/renderer/builder.lua [hunk 1]
```diff
@@ -209,15 +203,15 @@ function Builder:_build_file(node, padding, git_highlight, git_icons_tbl)
   local col_start = offset + #icon + git_icons_length
   local col_end = col_start + #node.name
 
-  if self.special_map[node.absolute_path] or self.special_map[node.name] then
+  if vim.tbl_contains(self.special_files, node.absolute_path) or vim.tbl_contains(self.special_files, node.name) then
```
Comment (kyazdani42) [inline lua/nvim-tree/renderer/builder.lua:206]: why not keeping existing implementation ?

**Review by kyazdani42 — APPROVED** [review]
looks good thank you !
Can you just answer this before we merge ?

File: lua/nvim-tree/renderer/builder.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree/renderer/builder.lua:206]: Do you mean the map of file names to boolean values?

I didn't see any reason for false value, so I made it a list like `window_picker.exclude.filetype`.

Comment (kyazdani42) [reply lua/nvim-tree/renderer/builder.lua:206]: well there is an issue then. Check out line 125 in the PR diff, special_files is used like a map, but here it is used as an array.

Comment (alex-courtis) [reply lua/nvim-tree/renderer/builder.lua:206]: Great catch!

`_build_folder` and `_build_file` are now the same, both tested OK.

**Review by kyazdani42 — APPROVED** [review]
LGTM !
Thanks a lot for this PR, we should have less issues coming up about people having issues with the setup !
Although we might have a few issues popping up in the next few days.
I'll leave you to merge this and update #674 (if you can, otherwise tell me i'll do it).

**Comment by alex-courtis** [comment]
#674 updated.

I do expect a couple of issues and am watchingi for issues.

**Comment by gegoune** [comment]
No issues on my side, config validator was extremely helpful! Great job Alex!

**Pull request merged**